saliency/tf1/base.py

```diff
@@ -130,23 +130,24 @@ def validate_xy_tensor_shape(self, x_steps=1, batch_size=1):
     # y should be shape (None,), (target_size), or () if target_size=1
     y_size = 1
     y_none_dim = False
-    for shape in self.y.shape:
+    y_shape = self.y.get_shape().as_list()
+    for shape in y_shape:
       if shape is None:
         y_none_dim = True
       else:
         y_size *= shape
```

Comment:
Yes the [2,3] case errors out later when values are returned. I guess the primary error this check is trying to catch is if users either (a) don't specify an individual class (i.e. class 236 of 1000) or (b) only select a single neuron when batching. In my understanding, for both of these cases calculating gradients will return the correct shape but will be an incorrect calculation. We can remove the checks and the code will still work, but users won't get back the right results. So if there's a better way to warn/check against that particular user error that'd be nice, but if we're constraining the user we could skip/loosen this validation step. What do you think?

Reply:
What if we simply check that the first dimension of 'y' satisfies the target_size instead of iterating through all shape elements? Regardless, the change is good; and it is more like a nit.